pages/star-search/index.tsx

```diff
@@ -301,97 +300,100 @@ export default function StarSearchPage({ userId, bearerToken, ogImageUrl }: Star
 
-        If the event type is function_call, we know we need to render a component, if the event type is content, we render the markdown as HTML.
+
+        If the content.type is "function_call", we know we need to render a component.
+        if the content.type is "content", we render the markdown as HTML.
        */
 
       const values = value.split("\n");
 
-      let eventType: EventType = "content";
-
       values.forEach(async (v) => {
         if (v.startsWith("id:")) {
-          return;
-        }
-
-        if (v.startsWith("event:")) {
-          eventType = v.split(/:\s+/)[1] as EventType;
+          // this is just the id of the SSE from the response.
           return;
         }
 
         if (v.startsWith("data:")) {
           /*
-           * regex for capturing star-search stream SSEs:
-           * data:\s?(?<result>.*)
+           * regex for capturing star-search stream events JSON:
+           * data:\s(?<result>.*)
            *
            * The aim of this regex is to capture all characters coming from
-           * the star-search server side events while also preserving the
-           * empty "data:" frames that may come through (which are newlines).
+           * the star-search server side events.
            *
            * 'data:' - matches the "data:" characters explicitly.
            * '\s'    - matches any whitespace that follows the data. In most cases, this is a single space ' '.
-           * '?'     - matches the previous whitespace token zero or one times. Aka, is optional.
+           *           So, for example, this captures 'data: '.
+           *
+           * '(?<result>.*)' - named capture group "result".
+           *    ├────── '?<result>'  - capture group is named "result".
+           *    └────── '.*'         - matches any characters (including zero characters) after the "data:\s" segment.
+           *                           Should capture ALL of the json object on the line after 'data: '.
            *
-           * '(?<result>.*)' - optional named capture group "result".
-           *    ├────── '?'          - capture group is optional.
-           *    ├────── '<result>'   - capture group is named "result".
-           *    └────── '.*'         - matches any characters (including zero characters) after the "data:\s?" segment.
-           *                           this is in service of also capturing empty strings as newlines.
+           * Example: data: { id: "abc123" }
+           * - Captures the 'data: ' (including the space)
+           * - The named capture group "result" gets the entire JSON object "{ id: \"abc123\" }"
+           *   as a string that can be parsed to an object.
            */
 
-          const matched = v.match(/data:\s?(?<result>.*)/);
+          const matched = v.match(/data:\s(?<result>.*)/);
```

Comment:
I made the regex _just abit_ more strict: now, there has to be a space between `data:` and the json object flowing in. We can always expect the same result now from the API: the `data` key, a `:`, a space ` `, and then the json object. So, for example, `data: { id: "abc123" }` will get captured. But `data:` or `data:{ ... }` will be rejected.

Reply:
always one space or can it be `\s+`? I'd assume just `\s` since the API is handling the formatting?

Reply:
Yes, correct, the payload from the API should always be `data: { ... }` with one space and we should reject anything that is malformed. It's probably a moot point to make the regex more strict, but it at least makes the client have to do _alot_ less heavy lifting since the API is doing it now